packages/components/src/TimeFormatter/TimeFormatter.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+
+interface TimeFormatterProps {
+  /**
+   * Civil Time string to be formatted
+   * 17:45:00.000000000
+   * 11:30:00.000000000
+   */
+  readonly civilTime: string;
```

Comment:
@kkimble006 Maybe not directly related to the work in your PR. I'm just curious why we're using `readonly` for props? I think I've seen it in other places too, but I'm pretty sure typescript makes props immutable (i.e. readonly) by default. 

Just to try it out, you can remove the `readonly` keyword, attempt to change the value using something like `props.civilTime = 'someOtherString'` and it will return a `TypeError`.

I'm trying to find some documentation that _guarantees_  props are readonly to confirm. Some starters:
https://reactjs.org/docs/components-and-props.html#props-are-read-only

![image](https://user-images.githubusercontent.com/12400201/62795453-08f96900-ba94-11e9-97ad-9a093fca487b.png)

@seanhealy @darryltec  Maybe you could also add some insight here?

Reply:
The readonly here is telling typescript that the property is readonly. This allows the compiler to find this error for us instead of relying on the codepath being executed and reported at runtime.

![image](https://user-images.githubusercontent.com/131327/63110231-29b83780-bf48-11e9-9d0a-a58fa12f0d12.png)

vs

![image](https://user-images.githubusercontent.com/131327/63110243-363c9000-bf48-11e9-9f36-c59da9a8c535.png)

It is nice to know that react is double checking this for us though too. :)